coderd/database/dbtestutil/db.go

```diff
@@ -51,3 +81,51 @@ func NewDB(t testing.TB) (database.Store, pubsub.Pubsub) {
 
 	return db, ps
 }
+
+// setRandDBTimezone sets the timezone of the database to the given timezone.
+// The timezone change does not take effect until the next session, so
+// we do this in our own connection
+func setRandDBTimezone(t testing.TB, dbURL, dbname string) {
+	t.Helper()
+
+	sqlDB, err := sql.Open("postgres", dbURL)
+	require.NoError(t, err)
+	defer func() {
+		_ = sqlDB.Close()
+	}()
+
+	// Pick a random timezone. We can simply pick from pg_timezone_names.
+	var tz string
+	err = sqlDB.QueryRow("SELECT name FROM pg_timezone_names ORDER BY RANDOM() LIMIT 1").Scan(&tz)
+	require.NoError(t, err)
+
+	// Set the timezone for the database.
+	t.Logf("setting timezone of database %q to %q", dbname, tz)
+	// We apparently can't use placeholders here, sadly.
+	// nolint: gosec // This is not user input and this is only executed in tests
+	_, err = sqlDB.Exec(fmt.Sprintf("ALTER DATABASE %s SET TIMEZONE TO %q", dbname, tz))
```

Comment:
self-review: I hate this but it appears necessary.

Reply:
It's fine in tests, IMO. It's an unfortunate limitation of some PostgreSQL queries.